### PR TITLE
Add a case about scalability test with max mem and two cpus

### PR DESCRIPTION
**Pull request by yiqianwei**
ID: 2033176

Signed-off-by: Yiqian Wei <yiwei@redhat.com>

### CODE REVIEW

**Comment by yiqianwei** [comment]
Run auto cmd:
python3 ConfigTest.py --testcase=boot_with_max_mem --guestname=RHEL.8.6.0 --customsparams="extra_params = -global ICH9-LPC.acpi-pci-hotplug-with-bridge-support=off"  --clone=no --debug
Auto results:
 (1/1) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.x86_64.io-github-autotest-qemu.boot_with_max_mem.q35: PASS (103.43 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2021-12-16T01.58-8a715b2/results.html
JOB TIME   : 105.49 s

**Comment by yiqianwei** [comment]
@jingzhao84, @yanan-fu Could you help to review this patch? Thanks

File: qemu/tests/cfg/boot_with_max_mem.cfg [hunk 1]
```diff
@@ -0,0 +1,7 @@
+- boot_with_max_mem:
+    virt_test_type = qemu
+    only Linux
+    type = boot_with_max_mem
+    start_vm = no
+    boot_mem = 8T
```
Comment (yanan-fu) [inline qemu/tests/cfg/boot_with_max_mem.cfg:6]: `mem_fixed = 8T`

File: qemu/tests/cfg/boot_with_max_mem.cfg [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+    boot_smp = 2
```
Comment (yanan-fu) [inline qemu/tests/cfg/boot_with_max_mem.cfg:7]:
```
smp_fixed = 2                                                                                                     
vcpu_maxcpus=${smp_fixed} 
```

File: qemu/tests/boot_with_max_mem.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    3) Check guest dmesg, no call trace message
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    host_mem = utils_misc.get_usable_memory_size() // 1024 // 1024
+    boot_mem = params.get("boot_mem")
+    if host_mem < int(boot_mem[0:-1]):
+        test.cancel("The host memory is less than %s of boot guest memory" % boot_mem)
```
Comment (yanan-fu) [inline qemu/tests/boot_with_max_mem.py:20]: If add param `vm_mem_minimum = 8T` in cfg, L17~L20 can be removed,  framework can handle it automatically, cancel test when no enough mem with host.

File: qemu/tests/cfg/boot_with_max_mem.cfg (hunk 2 above)
Comment (yanan-fu) [reply qemu/tests/cfg/boot_with_max_mem.cfg:7]: To be more strictly,  add the topology:
```
vcpu_cores = 0                                                                  
vcpu_threads = 1                                                                
vcpu_sockets = 2                                                                                                                                                
vcpu_dies = 1 
```

File: qemu/tests/cfg/boot_with_max_mem.cfg [hunk 4]
```diff
@@ -0,0 +1,7 @@
+- boot_with_max_mem:
+    virt_test_type = qemu
+    only Linux
```
Comment (yanan-fu) [inline qemu/tests/cfg/boot_with_max_mem.cfg:3]: From the manual case, it is x86_64 only.  
How about other arch ?   Does ppc64le, aarch64, s390x support max mem 8T ?

File: qemu/tests/boot_with_max_mem.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    params["mem"] = params.get("boot_mem")
+    params["smp"] = int(params.get("boot_smp"))
+    params["vcpu_maxcpus"] = int(params.get("boot_smp"))
```
Comment (yanan-fu) [inline qemu/tests/boot_with_max_mem.py:23]: No need, L21 ~L23

File: qemu/tests/cfg/boot_with_max_mem.cfg (hunk 2 above)
Comment (yiqianwei) [reply qemu/tests/cfg/boot_with_max_mem.cfg:7]: > ```
> smp_fixed = 2                                                                                                     
> vcpu_maxcpus=${smp_fixed} 
> ```
 I will update the code in the above way，thanks

File: qemu/tests/cfg/boot_with_max_mem.cfg (hunk 4 above)
Comment (yiqianwei) [reply qemu/tests/cfg/boot_with_max_mem.cfg:3]: > From the manual case, it is x86_64 only. 
yes

>How about other arch ? Does ppc64le, aarch64, s390x support max mem 8T ?
for multi-arch, please check following messages:
"For power, we had an agreement with IBM, they will cover limitations testing from their side.
For z systems, we don't have such a big machine in the pool.
For arm, it's not official support yet and we have very limited resources, QE now only focus on sanity testing. And we don't have big machines with huge mem/cpu as well."

File: qemu/tests/cfg/boot_with_max_mem.cfg (hunk 1 above)
Comment (yiqianwei) [reply qemu/tests/cfg/boot_with_max_mem.cfg]: @yanan-fu , I have updated it, please check it, thanks

File: qemu/tests/boot_with_max_mem.py (hunk 3 above)
Comment (yiqianwei) [reply qemu/tests/boot_with_max_mem.py]: @yanan-fu , I already add param vm_mem_minimum = 8T in cfg, L17~L20 can be removed, please check it, thanks

File: qemu/tests/boot_with_max_mem.py (hunk 5 above)
Comment (yiqianwei) [reply qemu/tests/boot_with_max_mem.py]: @yanan-fu , I have removed L21 ~ L23, please check it, thanks

File: qemu/tests/cfg/boot_with_max_mem.cfg (hunk 4 above)
Comment (yiqianwei) [reply qemu/tests/cfg/boot_with_max_mem.cfg:3]: @yanan-fu , I already added  "only x86_64" to cfg file,please check it ,thanks

File: qemu/tests/cfg/boot_with_max_mem.cfg (hunk 2 above)
Comment (yiqianwei) [reply qemu/tests/cfg/boot_with_max_mem.cfg]: @yanan-fu, The update has been completed, please check it

**Comment by yiqianwei** [comment]
@yanan-fu,I have updated this patch, please review this patch again. thanks

Run auto cmd: python3 ConfigTest.py --testcase=boot_with_max_mem --machines=q35 --guestname=RHEL.8.6.0 --clone=no --debug
Auto results:
(1/1) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.x86_64.io-github-autotest-qemu.boot_with_max_mem.q35: PASS (97.04 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2022-01-03T23.18-bbd4f18/results.html

**Comment by jingzhao84** [comment]
@yanan-fu 
Hello, could you help to review this patch, many thanks
cc @yiqianwei

File: qemu/tests/boot_with_max_mem.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    :param env: Dictionary with test environment.
+    """
+    params["start_vm"] = "yes"
+    vm_name = params["main_vm"]
+    env_process.preprocess_vm(test, params, env, vm_name)
+    vm = env.get_vm(vm_name)
+    vm.verify_alive()
+    timeout = float(params.get("login_timeout", 360))
+    session = vm.wait_for_login(timeout=timeout)
+    utils_misc.verify_dmesg(level_check=4, session=session)
```
Comment (yanan-fu) [inline qemu/tests/boot_with_max_mem.py:24]: Suggest `vm.verify_dmesg`  and set suitable `guest_dmesg_level`

Comment (yiqianwei) [reply qemu/tests/boot_with_max_mem.py:24]: The return result of verify_dmesg() is different from the expected result, so verify_dmesg() cannot be used, I will change the code in other ways, thank you

**Comment by yiqianwei** [comment]
@yanan-fu ,could you help to review this patch,thanks a lot

**Comment by yiqianwei** [comment]
discuss with yfu and jinzhao on online meeting,  boot case with custom params can cover, so close this patch,thanks

**Pull request closed without merging**